src/mysql_protocol.erl

```diff
@@ -1078,6 +1078,18 @@ encode_param(Value) when is_integer(Value), Value < 0 ->
     end;
 encode_param(Value) when is_float(Value) ->
     {<<?TYPE_DOUBLE, 0>>, <<Value:64/float-little>>};
+encode_param({decimal, Value}) ->
+    Bin = if  is_binary(Value) ->
+                  Value;
+              is_list(Value) ->
+                  list_to_binary(Value);
+              is_integer(Value) ->
+                  integer_to_binary(Value);
+              is_float(Value) ->
+                  list_to_binary(io_lib:format("~w", [Value]))
+          end,
+    EncLength = lenenc_int_encode(byte_size(Bin)),
+    {<<?TYPE_DECIMAL, 0>>, <<EncLength/binary, Bin/binary>>};
```

Comment:
Alternatively, maybe do it with multiple function clauses like this?

```erlang
...
encode_param({decimal, Bin}) when is_binary(Bin) ->
    EncLength = lenenc_int_encode(byte_size(Bin)),
    {<<?TYPE_DECIMAL, 0>>, <<EncLength/binary, Bin/binary>>};
encode_param({decimal, List}) when is_list(List) ->
    encode_param({decimal, list_to_binary(List)});
...
```

In case of an invalid type for `Value`, this would result in a `function_clause` instead of an `if_clause` error.

Reply:
It looks clean but re-wrapping it in a tuple again is unnecessary work. It doesn't matter which error it crashes with. (Perhaps I can just skip the likebreaks after `->` in the if clauses so it looks more compact?)

Reply:
In that case, I'd leave it as it is. Removing the linebreaks would make it hard to read IMO.